### PR TITLE
Bug 1825429 - Tapping on empty search result should cancel search mode.

### PR DESCRIPTION
**Pull request by iorgamgabriel**
### Pull Request checklist

- [x] **Quality**: This PR builds and passes detekt/ktlint checks (A pre-push hook is recommended)
- [ ] **Tests**: This PR includes thorough tests or an explanation of why it does not
- [ ] **Changelog**: This PR includes [a changelog entry](https://github.com/mozilla-mobile/firefox-android/blob/main/docs/changelog.md) or does not need one
- [x] **Accessibility**: The code in this PR follows [accessibility best practices](https://github.com/mozilla-mobile/firefox-android/blob/main/docs/shared/android/accessibility_guide.md) or does not include any user facing features

### After merge
- **Breaking Changes**: If this is a breaking Android Components change, please push a draft PR on [Reference Browser](https://github.com/mozilla-mobile/reference-browser) to address the breaking issues.

### To download an APK when reviewing a PR (after all CI tasks finished running):
1. Click on `Checks` at the top of the PR page.
2. Click on the `firefoxci-taskcluster` group on the left to expand all tasks.
3. Click on the `build-apk-{fenix,focus,klar}-debug` task you're interested in.
4. Click on `View task in Taskcluster` in the new `DETAILS` section.
5. The APK links should be on the right side of the screen, named for each CPU architecture.

### GitHub Automation
https://bugzilla.mozilla.org/show_bug.cgi?id=1825429

### CODE REVIEW

**Comment by iorgamgabriel** [comment]
https://user-images.githubusercontent.com/8118371/230607163-80d8c5f0-c518-4d64-88f9-0c677b50db35.mp4

File: fenix/app/src/main/java/org/mozilla/fenix/search/SearchDialogFragment.kt [hunk 1]
```diff
@@ -338,7 +338,11 @@ class SearchDialogFragment : AppCompatDialogFragment(), UserInteractionHandler {
         when (getPreviousDestination()?.destination?.id) {
             R.id.browserFragment, R.id.homeFragment -> {
                 binding.searchWrapper.setOnTouchListener { _, _ ->
-                    binding.searchWrapper.hideKeyboard()
+                    if (toolbarView.view.url.isEmpty()) {
+                        dismissAllowingStateLoss()
```
Comment (Alexandru2909) [inline fenix/app/src/main/java/org/mozilla/fenix/search/SearchDialogFragment.kt:342]: I think adding the dismiss call here breaks the interaction with the home fragment private [button](https://searchfox.org/mozilla-mobile/source/firefox-android/fenix/app/src/main/java/org/mozilla/fenix/search/SearchDialogFragment.kt#273): 

https://user-images.githubusercontent.com/35462038/234011207-c43d6a22-1db3-4f70-adea-b06cbd4ba322.mp4

**Review by Alexandru2909 — APPROVED** [review]
LGTM! It looks like we've reached the limit for detekt's `ComplexMethod` in `onViewCreated`. Let's suppress that and this should be good to land.

**Comment by iorgamgabriel** [comment]
@mergify backport releases_v113

**Pull request merged**